### PR TITLE
[SPARK-10251][CORE] some common types are not registered for Kryo Serializat…

### PR DESCRIPTION
**Pull request by harsha2010**
…ion by default

### CODE REVIEW

**Comment by harsha2010** [comment]
@rxin can you please take a look?

File: core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala [hunk 1]
```diff
@@ -131,6 +132,35 @@ class KryoSerializer(conf: SparkConf)
     // our code override the generic serializers in Chill for things like Seq
     new AllScalaRegistrar().apply(kryo)
 
+    // Register types missed by Chill.
+    kryo.register(classOf[Array[Tuple1[Any]]])
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala]: would this register (Int, Int) since they are specialized?

Comment (harsha2010) [inline core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala]: @rxin  yes, there is a check for (Int, Int) in the KryoSuite and without registering Array[Tuple2[Any, Any]] that test fails.

**Comment by harsha2010** [comment]
jenkins, please test

**Comment by rxin** [comment]
Thanks - I've merged this.

**Pull request closed without merging**